addons/test_event_full/tests/test_event_mail.py

```diff
@@ -1,15 +1,52 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from datetime import timedelta
+from datetime import datetime, timedelta
 
+from odoo import fields
 from odoo.addons.mail.tests.common import MockEmail
 from odoo.addons.sms.tests.common import MockSMS
+from odoo.addons.test_event_full.tests.common import TestEventFullCommon
 from odoo.addons.test_event_full.tests.common import TestWEventCommon
 from odoo.exceptions import ValidationError
 from odoo.tools import mute_logger
 
-class TestTemplateRefModel(TestWEventCommon):
+class TestTemplateRefModel(TestEventFullCommon, TestWEventCommon):
+
+    def test_template_ref_delete_lines(self):
+        """ When deleting a template, related lines should be deleted too """
+
+        self.event_type_complex.write({
+            'event_type_mail_ids': [
+                (0, 0, {
+                    'interval_unit': 'now',
+                    'interval_type': 'after_sub',
+                    'notification_type': 'sms',
+                    'template_ref': 'sms.template,%i' % self.env['ir.model.data']._xmlid_to_res_id('event_sms.sms_template_data_event_registration')}),
+            ],
+        })
+
+        template_mail = self.event_type_complex.event_type_mail_ids[0].template_ref
+        template_sms = self.event_type_complex.event_type_mail_ids[2].template_ref
+
+        event = self.env['event.event'].create({
+            'name': 'event mail template removed',
+            'event_type_id': self.event_type_complex.id,
+            'auto_confirm': True,
+            'date_begin': fields.Datetime.to_string(datetime.today() + timedelta(days=1)),
+            'date_end': fields.Datetime.to_string(datetime.today() + timedelta(days=15)),
```

Comment:
Normally you can just feed that in as a datetime and the field will do this on its own.
Would avoid importing fields too.

Also could use a fixed datetime both times as it's not relevant and it makes it clear there's no risk of the test randomly failing due to the date.

Reply:
Done